video_xblock/video_xblock.py

```diff
@@ -698,7 +743,44 @@ def update_metadata_authentication(self, auth_data, player):
             self.metadata['access_token'] = ''   # Brightcove API
             self.metadata['client_id'] = ''      # Brightcove API
             self.metadata['client_secret'] = ''  # Brightcove API
-        # Clear metadata (only backend-specific parameters are to be stored)
-        irrelevant_fields = [key for key in self.metadata if key not in player.metadata_fields]
-        for key in irrelevant_fields:
-            del self.metadata[key]
+
+    @XBlock.json_handler
+    def upload_default_transcript_handler(self, data, suffix=''):  # pylint: disable=unused-argument
+        """
+        Function for uploading a transcript fetched a video platform's API to video xblock.
+
+        """
+        player = self.get_player()
+        video_id = player.media_id(self.href)
+        lang_code = str(data.get(u'lang'))
+        lang_label = str(data.get(u'label'))
+        sub_url = str(data.get(u'url'))
+        # File name format is <language label>_captions_video_<video_id>, e.g. "English_captions_video_456g68"
+        reference_name = "{}_captions_video_{}".format(lang_label, video_id).encode('utf8')
+
+        # Fetch default transcript
+        sub_unicode = player.download_default_transcript(url=sub_url, language_code=lang_code)
+        sub = self.convert_caps_to_vtt(caps=sub_unicode)
+
+        # Define location of default transcript as a future asset and prepare content to store in assets
+        ext = '.vtt'
+        file_name = reference_name.replace(" ", "_") + ext
+        course_key = self.location.course_key  # pylint: disable=no-member
+        content_loc = StaticContent.compute_location(course_key, file_name)  # AssetLocator object
+        sc_partial = functools.partial(StaticContent, content_loc, file_name, 'application/json')
+        content = sc_partial(sub.encode('UTF-8'))  # StaticContent object
+        external_url = '/' + str(content_loc)
+
+        # Commit the content
+        # TODO add additional checks before saving to contenstore as per the reference https://git.io/vDnBN
```

Comment:
Since we can not delete cached thumbnail because of `del_cached_content()` import error, for now I'd prefer to simply delete this TODO and save conent to contentstore without a `thumbnail_location` (to avoid possible errors and testing issues)